### PR TITLE
refactor(core): Remove custom event and replay behavior.

### PR DESCRIPTION
**Pull request by tbondwilkinson**
These behaviors have been moved back to g3.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

### CODE REVIEW

**Comment by tbondwilkinson** [comment]
TESTED=[TGP](https://fusion2.corp.google.com/presubmit/631197094/OCL:631197094:BASE:631874495:1715194028137:1d0af9b6/targets)

**Comment by tbondwilkinson** [comment]
Caretaker note:
- Copybara CL required to create sync CL: cl/631196825
- BUILD changes required to patch into sync CL: 631207106

File: packages/core/primitives/event-dispatch/test/dispatcher_test.ts [hunk 1]
```diff
@@ -219,7 +218,7 @@ describe('dispatcher test.ts', () => {
     dispatcher.registerGlobalHandler('click', handler);
 
     const eventInfo = createTestEventInfo({
-      event: createEvent({type: 'mousedown'} as Event),
+      event: new MouseEvent('mousedown', {bubbles: true, cancelable: true}),
```
Comment (thePunderWoman) [inline packages/core/primitives/event-dispatch/test/dispatcher_test.ts:221]: I think we'll need to add support for Blossom and Buttercup, too.

**Review by rahatarmanahmed — APPROVED** [review]

**Review by thePunderWoman — APPROVED** [review]
LGTM

reviewed-for: public-api, primitives, size-tracking

File: goldens/size-tracking/integration-payloads.json [hunk 2]
```diff
@@ -56,7 +56,7 @@
     "uncompressed": {
       "main": 200584,
       "polyfills": 33807,
-      "event-dispatch-contract.min": 11293
+      "event-dispatch-contract.min": 10513
```
Comment (AndrewKushnir) [inline goldens/size-tracking/integration-payloads.json:59]: 🎉

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
@tbondwilkinson this PR conflicts with the `main` branch (a golden file needs an update) after merging https://github.com/angular/angular/pull/55587. Could you please rebase and remove the "cleanup" label from this PR when you get a chance?

**Comment by tbondwilkinson** [comment]
> @tbondwilkinson this PR conflicts with the `main` branch (a golden file needs an update (likely a revert)) after merging #55587. Could you please rebase and remove the "cleanup" label from this PR when you get a chance?

Done!

**Comment by atscott** [comment]
This PR was merged into the repository by commit 76d0aea89b1d34a90761d15829cd0c0ca6b44726.

**Pull request closed without merging**